src/libs/actions/Policy.js

```diff
@@ -809,10 +809,7 @@ function createWorkspace() {
         expenseReportActionData,
     } = Report.buildOptimisticWorkspaceChats(policyID, workspaceName);
 
-    // We need to use makeRequestWithSideEffects as we try to redirect to the policy right after creation
-    // The policy hasn't been merged in Onyx data at this point, leading to an intermittent Not Found screen
-    // eslint-disable-next-line rulesdir/no-api-side-effects-method
-    API.makeRequestWithSideEffects('CreateWorkspace', {
+    API.write('CreateWorkspace', {
```

Comment:
`API.write` is the way to go. We used `makeRequestWithSideEffects` because otherwise you get a brief "Not found" page when creating the workspace. That still happens, and you should add this creation flow from another workspace to the tests. Here's a video showing the behavior. It's very brief after your `policyIDListExcludingWorkspacesCreatedOffline` changes, but still noticeable.

https://user-images.githubusercontent.com/26260477/191348741-1f8dc40b-a0ad-4ca2-8649-c2166dcf1338.mov


If you change all optimistic data to use `SET` then it works very well since it's set is faster than merge.

https://user-images.githubusercontent.com/26260477/191348790-5b036a7a-4530-479e-b494-910b451fd50b.mov